openshift/openshift.go

```diff
@@ -396,17 +396,17 @@ func (d *openshiftImageDestination) ReapplyBlob(info types.BlobInfo) (types.Blob
 // FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
 // If the destination is in principle available, refuses this manifest type (e.g. it does not recognize the schema),
 // but may accept a different manifest type, the returned error must be an ManifestTypeRejectedError.
-func (d *openshiftImageDestination) PutManifest(m []byte) error {
+func (d *openshiftImageDestination) PutManifest(m []byte, instanceDigest *digest.Digest) error {
 	manifestDigest, err := manifest.Digest(m)
 	if err != nil {
 		return err
 	}
 	d.imageStreamImageName = manifestDigest.String()
 
-	return d.docker.PutManifest(m)
```

Comment:
Above, `d.imageStreamImageName` should only be recorded for the primary instance.

Reply:
Oh, right.  That makes sense.  Fixed.

Reply:
The `d.imageStreamImageName = manifestDigest.String()` assignment is still unconditional AFAICS.